sdk/core/core/src/az_span.c

```diff
@@ -129,6 +129,78 @@ AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number)
   return AZ_OK;
 }
 
+AZ_NODISCARD az_result az_span_find(az_span source, az_span target, az_span* out_span)
+{
+  AZ_PRECONDITION_VALID_SPAN(source, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(target, 1, false);
+  AZ_PRECONDITION_NOT_NULL(out_span);
+
+  uint8_t* source_ptr = az_span_ptr(source);
+  int32_t source_length = az_span_length(source);
+  uint8_t* target_ptr = az_span_ptr(target);
+  int32_t target_length = az_span_length(target);
+
+  for (int i = 0; i < source_length; i++)
+  {
+    if (source_ptr[i] == target_ptr[0])
+    {
+      int j;
+      for (j = 1; j < target_length && (i + j) < source_length; j++)
+      {
+        if (source_ptr[i + j] != target_ptr[j])
+        {
+            break;
+        }
+      }
+
+      if (j == target_length)
+      {
+        *out_span = az_span_init(source_ptr + i, target_length, target_length);
+
+        return AZ_OK;
+      }
+    }
+  }
+
+  return AZ_ERROR_EOF;
+}
+
+AZ_NODISCARD az_result az_span_token_next(az_span source, az_span delimiter, az_span* out_token, az_span* out_span)
+{
+  AZ_PRECONDITION_VALID_SPAN(source, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(delimiter, 1, false);
+  AZ_PRECONDITION_NOT_NULL(out_token);
+  AZ_PRECONDITION_NOT_NULL(out_span);
+
+  az_span instance;
```

Comment:
nit: I'd move this to just before the if statement. It is good practice to not allow uninitialized variables to live long in the code. That is declare them right next to initialization. Someone else working on this code may attempt to use instance between lines 175 & 180 resulting in unpredictable behavior. Declaring just before initialization decreases code brittleness.

Reply:
Sure.